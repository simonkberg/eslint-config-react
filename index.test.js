/* eslint-env jest */

'use strict'

const fs = require('fs')
const path = require('path')
const config = require('./index.js')
const { CLIEngine } = require('eslint')

expect.addSnapshotSerializer({
  print: val => JSON.stringify(val.replace(__dirname, '<PROJECT_ROOT>')),
  test: val => typeof val === 'string',
})

describe('@simonkberg/eslint-config-react', () => {
  it('should match the snapshot', () => expect(config).toMatchSnapshot())

  it('should load config in eslint', done => {
    const cli = new CLIEngine({
      ignore: false,
      useEslintrc: false,
      configFile: path.resolve(__dirname, 'index.js'),
    })

    fs.readFile(
      path.resolve(__dirname, '__fixtures__/react.js'),
      'utf-8',
      (err, file) => {
        expect(err).toBeNull()

        const { errorCount, warningCount } = cli.executeOnText(file)

        expect(errorCount).toBe(0)
        expect(warningCount).toBe(0)
        done()
      }
    )
  })
})
