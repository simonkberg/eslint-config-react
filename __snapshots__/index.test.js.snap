// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@simonkberg/eslint-config-react should match the snapshot 1`] = `
Object {
  "extends": Array [
    "<PROJECT_ROOT>/node_modules/@simonkberg/eslint-config/index.js",
    "<PROJECT_ROOT>/node_modules/@simonkberg/eslint-config/react.js",
  ],
}
`;
